### ISSUE TITLE
Possibility of supporting Folly on Mac OS X?

### ISSUE THREAD

**Issue by c9s** [facebook/folly#93]
Currently there are some non-portable items that breaks the build in HHVM, Just wondering the possibility of the Mac OS X fix and the PR you intend to receive.

Thanks!

```
In file included from /Volumes/ramdisk/hhvm/third-party/folly/folly/Baton.h:26:0,
                 from /Volumes/ramdisk/hhvm/third-party/folly/folly/LifoSem.h:28,
                 from /Volumes/ramdisk/hhvm/third-party/folly/folly/experimental/wangle/concurrent/LifoSemMPMCQueue.h:19,
                 from /Volumes/ramdisk/hhvm/third-party/folly/folly/experimental/wangle/concurrent/ThreadPoolExecutor.h:19,
                 from /Volumes/ramdisk/hhvm/third-party/folly/folly/experimental/wangle/concurrent/CPUThreadPoolExecutor.h:18,
                 from /Volumes/ramdisk/hhvm/third-party/folly/folly/experimental/wangle/concurrent/CPUThreadPoolExecutor.cpp:17:
/Volumes/ramdisk/hhvm/third-party/folly/folly/detail/Futex.h:24:25: fatal error: linux/futex.h: No such file or directory
 #include <linux/futex.h>
                         ^
compilation terminated.
make[2]: *** [third-party/folly/CMakeFiles/folly.dir/folly/experimental/wangle/concurrent/CPUThreadPoolExecutor.cpp.o] Error 1
make[1]: *** [third-party/folly/CMakeFiles/folly.dir/all] Error 2
```

**Comment by facebook-github-bot-9** [comment]
Thank you for reporting this issue and appreciate your patience. We've  notified the core team for an update on this issue. We're looking for a response within the next 30 days or the issue may be closed.

**Comment by chipturner** [comment]
Thank you for your patience. We're going to close out this issue. If you're still experiencing the problem described, please reopen the issue - thanks!

**Comment by chipturner** [comment]
Hello!  This is a human.  My previous comment on this issue was a robot running amok.  I apologize that the issue was closed, and am re-opening it.  Thanks for your help with our folly project and for your understanding about rogue robots trying to take over the world.

**Comment by sgolemon** [comment]
I'm going to close this task since HHVM is actually building on MacOSX right now (yes, I know we haven't released packages yet), but it doesn't need to exist as a meta-task.  The CI for Folly/HHVM on Mac is also coming online.

TL;DR - You'll get Mac support and it'll *stay* working.
